platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/parameters/BuildProgressListenerAdapter.java

```diff
@@ -612,14 +612,45 @@ private TestOutputEvent transformTestOutput(InternalTestOutputEvent event, Inter
         return null;
     }
 
-    private static List<ProblemSummary> toProblemIdSummaries(List<InternalProblemSummary> problemIdCounts) {
+    @Nonnull
+    static List<ProblemSummary> toProblemIdSummaries(List<InternalProblemSummary> problemIdCounts) {
+        Map<ProblemId, List<InternalProblemSummary>> groupedSummaries = getGroupedMap(problemIdCounts);
```

Comment:
This is the meat of this PR.
getting all summaries with the same id and the summing up the counts.
With that we can in future extend the ProblemSummary but still keep the values for older clients the same.